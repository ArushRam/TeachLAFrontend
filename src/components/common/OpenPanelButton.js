import React from "react";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";

/**-------Props--------
 * panelOpen: boolean, is profile panel visible
 * togglePanel: function to call when you want the Profile Panel to disappear/reappear
 */

const OpenPanelButton = ({ panelOpen, togglePanel }) => {
  //if the left panel is closed, show nothing
  //otherwise show hamburger icon
  return (
    <div className="editor-expand-panel-arrow" title="Open Profile Panel" onClick={togglePanel}>
      {panelOpen ? "" : <FontAwesomeIcon icon="bars" />}
    </div>
  );
};

export default OpenPanelButton;
