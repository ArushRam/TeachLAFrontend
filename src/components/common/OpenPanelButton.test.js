import React from "react";
import { shallow } from "enzyme";
import OpenPanelButton from "./OpenPanelButton";

const clickFn = jest.fn();

describe("LoadingPage", () => {
  it("panelOpen=true snapshot", () => {
    const component = shallow(<OpenPanelButton panelOpen={true} />);

    expect(component).toMatchSnapshot();
  });

  it("panelOpen=false snapshot", () => {
    const component = shallow(<OpenPanelButton panelOpen={false} />);

    expect(component).toMatchSnapshot();
  });

  it("handles undefined togglePanel prop", () => {
    const component = shallow(<OpenPanelButton panelOpen={false} />);

    component.find("div#editor-expand-panel-arrow", {}, results => {
      expect(results.length).toBe(1);
      results[0].simulate("click");
    });
  });

  it("calls togglePanel on click", () => {
    const component = shallow(<OpenPanelButton panelOpen={false} togglePanel={clickFn} />);

    component.find("div#editor-expand-panel-arrow").simulate("click");

    expect(clickFn).toHaveBeenCalled();
  });
});
