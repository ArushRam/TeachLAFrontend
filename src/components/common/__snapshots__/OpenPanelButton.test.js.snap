// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingPage handles undefined panelOpen snapshot 1`] = `ShallowWrapper {}`;

exports[`LoadingPage panelOpen=false snapshot 1`] = `ShallowWrapper {}`;

exports[`LoadingPage panelOpen=true snapshot 1`] = `ShallowWrapper {}`;
